teps/0101-decouple-catalog-organization-and-reference.md

```diff
@@ -0,0 +1,233 @@
+---
+status: proposed
+title: Decouple catalog organization and reference
+creation-date: '2022-03-21'
+last-updated: '2022-03-21'
+authors:
+- "@vdemeester"
+---
+
+# TEP-0101: Decouple catalog organization and resource usage
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes/Caveats (optional)](#notescaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [Implementation Pull request(s)](#implementation-pull-request-s)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+As described in [TEP-0003](./0003-tekton-catalog-organization.md) :
+The Tekton catalog is a collection of blessed Tekton resources that
+can be used with any system that supports the Tekton API. We also have
+the [hub](https://hub.tekton.dev) to be able to search for Tasks.
+
+This proposes to make the *Hub* and any other **official** mean of
+consuming Tekton resources from the community decoupled from where we
+author them. In a gist, this aims to make the repository
+`tektoncd/catalog` an implementation detail, and only support
+long-term, non-changing URI to refer to tasks (Those URIs could be
+`https://` endpoint such as the Hub or docker images, …).
+
+## Motivation
+
+As of today, users of the tekton task are directly refering to the
+GitHub URL to get Task from the catalog. This creates a hard coupling
+between the way we organize tasks inside Tekton and the way user
+consume it. This is forcing us into one possible catalog organization
+and prevent us to make drastic changes to this organization without
+breaking users.
+
+This enhancement aim to decouple the catalog organization from users
+consumption of those resources. This enhancement proposal is mainly
+touching the catalog and the hub components. This would also push the
+hub to be the main entrypoint for users to search, list and get tekton
+resources.
+
+### Goals
+
+- Allowing for the maintainers of `tektoncd/catalog` to change the
+  organization without affecting the consumption of the resources.
```

Comment:
We might be able to do this for newer version of tasks and pipelines but I don't think we should event remove old versions in their current format